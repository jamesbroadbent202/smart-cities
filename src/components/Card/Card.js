import React from 'react';
import PropTypes from 'prop-types';
import style from './Card.scss';
import {
  CARD_SIZES,
  INDICATOR_CARD_TYPES,
} from '../../constants';

const classnames = require('classnames/bind').bind(style);

// This component is not meant to be used directly.
// Use either a <CategoryCard> or an <IndicatorCard>
const Card = (props) => {
  const className = classnames(
    'card',
    `card__${props.size}`,
    { card__category: props.type === INDICATOR_CARD_TYPES.CATEGORY },
    props.className,
  );

  return (
    <div
      className={className}
      style={{ borderBottomColor: props.color }}
    >
      {props.children}
    </div>
  );
};

Card.propTypes = {
  children: PropTypes.oneOfType([
    PropTypes.node,
    PropTypes.array,
  ]).isRequired,
  className: PropTypes.string,
  color: PropTypes.string.isRequired,
  size: PropTypes.oneOf(Object.values(CARD_SIZES)).isRequired,
  type: PropTypes.oneOf(Object.values(INDICATOR_CARD_TYPES)).isRequired,
};

export default Card;
