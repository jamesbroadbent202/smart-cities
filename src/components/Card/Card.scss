@import '~style/variables';

.card {
  @include shadow;

  background: $color-white;
  border-bottom-style: solid;
  border-radius: 4px;
}

.card__small {
  height: 80px;
  padding: 4px 8px;
  border-bottom-width: 6px;
}

.card__large {
  height: 160px;
  padding: 8px 16px;
  border-bottom-width: 8px;
}

.card__category {
  color: $color-white;
}
