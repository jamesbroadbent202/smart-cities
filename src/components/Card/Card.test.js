import React from 'react';
import { shallow } from 'enzyme';
import Card from './Card';
import {
  CARD_SIZES,
  INDICATOR_CARD_TYPES,
} from '../../constants';

const defaultProps = {
  size: CARD_SIZES.LARGE,
  type: INDICATOR_CARD_TYPES.INDICATOR,
  color: '#fff',
};

it('should match Snapshot', () => {
  const component =  shallow(
    <Card {...defaultProps}>
      <div>Render me</div>
    </Card>
  );

  expect(component.debug()).toMatchSnapshot();
});

it('should apply the correct size class', () => {
  const component =  shallow(
    <Card
      {...defaultProps}
      size={CARD_SIZES.SMALL}
    >
      <div>Render me</div>
    </Card>
  );

  expect(component.hasClass('card__small')).toBe(true);
});

it('should apply the correct color', () => {
  const component =  shallow(
    <Card
      {...defaultProps}
      color="#a11"
    >
      <div>Render me</div>
    </Card>
  );

  const componentStyle = component.prop('style');
  expect(componentStyle.borderBottomColor).toBe('#a11');
});

it('should apply the category class when the type is category', () => {
  const component =  shallow(
    <Card
      {...defaultProps}
      type={INDICATOR_CARD_TYPES.CATEGORY}
    >
      <div>Render me</div>
    </Card>
  );

  expect(component.hasClass('card__category')).toBe(true);
});
