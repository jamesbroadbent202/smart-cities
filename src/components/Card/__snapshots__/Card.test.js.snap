// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match Snapshot 1`] = `
"<div className=\\"card card__large\\" style={{...}}>
  <div>
    Render me
  </div>
</div>"
`;
